src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
+
+    @middleware.receive.execute(
+      {response: new Response(this, message)}
```

Comment:
> We could change the model so that a message is considered a request and there is a single response object shared across all matching listeners.

This results in weirdness inside listeners. `response.match` will be unique to each listener, but `response.message` would be shared. It also means that an early listener can change a message to no longer match another listener. I'm not sure that's an interaction I like; I feel like message modification that affects listeners should be the job of a receive middleware.

>  Alternatively, we could decide that the Express request/response model doesn't work well for us and define something completely different.

In chat, one "request" can fire multiple listeners and create multiple responses, a pretty different model than how HTTP works. Express isn't bad, and if we can be consistent, that's good. But I have a feeling that API will feel pretty awkward.